src/solver.f90

```diff
@@ -111,35 +109,44 @@ function init(backend, mesh, host_allocator) result(solver)
     solver%v => solver%backend%allocator%get_block(DIR_X)
     solver%w => solver%backend%allocator%get_block(DIR_X)
 
-    call read_solver_input(Re, dt, n_iters, n_output, poisson_solver_type, &
-                           time_intg, der1st_scheme, der2nd_scheme, &
-                           interpl_scheme, stagder_scheme)
+    if (command_argument_count() >= 1) then
+      call get_command_argument(1, input_file)
```

Comment:
I think this logic of reading command line argument should be defined elsewhere (maybe inside the config module?) because the code is duplicated also in `xcompact.f90` at the moment. Right now it isn't a big deal but we might be something to think about for when we want to add other command line arguments, a help page etc.

Reply:
Completely agree, not the best place. In general I think its better to avoid getting command arguments inside a module instead of from a program. For example test programs can still use these modules but in addition they may have their own stuff from the command line input, it would be hard to manage. We may instead read the solver namelist in the main program (we actually already do to get poisson solver) and pass the `solver_config_t` instance to solver, but not sure how this'll work when we have even more options/namelists. We may also deal with this later on and not in this PR.

Reply:
How about replacing this `if ...` with `call get_config()` which then runs this code? This could be defined in the `configuration` module and used/not used as required.

Reply:
I think this addresses both concerns?